packages/mui-material/src/TextField/TextField.test.js

```diff
@@ -244,12 +228,22 @@ describe('<TextField />', () => {
     });
   });
 
-  it('should trigger `onClick` only once', () => {
-    const handleClick = spy();
-    const { getByRole } = render(
-      <TextField variant="outlined" label="Test" onClick={handleClick} />,
-    );
-    fireEvent.click(getByRole('textbox'));
-    expect(handleClick.callCount).to.equal(1);
+  describe('event: click', () => {
+    it('registers `onClick` on the root slot', () => {
+      const handleClick = spy((event) => event.currentTarget);
+      const { getByTestId, getByRole } = render(
+        <TextField data-testid="root" onClick={handleClick} />,
+      );
+
+      const input = getByRole('textbox');
+
+      const root = getByTestId('root');
+
+      fireEvent.click(input);
+
+      expect(handleClick.callCount).to.equal(1);
+      // return value is event.currentTarget
+      expect(handleClick.returned(root));
```

Comment:
This test doesn't work, it doesn't fail without the fix.

This would be correct:

```suggestion
      expect(handleClick.returned(root)).to.equal(true);
```